Source/WebKit/UIProcess/mac/LegacySessionStateCoding.cpp

```diff
@@ -477,28 +480,31 @@ static RetainPtr<CFDictionaryRef> encodeSessionHistory(const BackForwardListStat
     return createDictionary({ { sessionHistoryVersionKey, sessionHistoryVersionNumber.get() }, { sessionHistoryCurrentIndexKey, currentIndexNumber.get() }, { sessionHistoryEntriesKey, entries.get() } });
 }
 
-RefPtr<API::Data> encodeLegacySessionState(const SessionState& sessionState)
+RefPtr<API::Data> encodeLegacySessionState(const SessionState& sessionState, std::optional<HashMap<WebCore::ClientOrigin, HashMap<String, String>>>&& sessionStorage)
 {
     auto sessionHistoryDictionary = encodeSessionHistory(sessionState.backForwardListState);
     auto provisionalURLString = sessionState.provisionalURL.isNull() ? nullptr : sessionState.provisionalURL.string().createCFString();
     RetainPtr<CFNumberRef> renderTreeSizeNumber(adoptCF(CFNumberCreate(kCFAllocatorDefault, kCFNumberSInt64Type, &sessionState.renderTreeSize)));
     RetainPtr<CFBooleanRef> isAppInitiated = adoptCF(sessionState.isAppInitiated ? kCFBooleanTrue : kCFBooleanFalse);
 
-    RetainPtr<CFDictionaryRef> stateDictionary;
-    if (provisionalURLString) {
-        stateDictionary = createDictionary({
-            { sessionHistoryKey, sessionHistoryDictionary.get() },
-            { provisionalURLKey, provisionalURLString.get() },
-            { renderTreeSizeKey, renderTreeSizeNumber.get() },
-            { isAppInitiatedKey, isAppInitiated.get() }
-        });
-    } else {
-        stateDictionary = createDictionary({
-            { sessionHistoryKey, sessionHistoryDictionary.get() },
-            { renderTreeSizeKey, renderTreeSizeNumber.get() },
-            { isAppInitiatedKey, isAppInitiated.get() }
-        });
-    }
+    RetainPtr<CFDictionaryRef> sessionStorageDictionary;
+    // FIXME: Implement encodeSessionStorage() to create a CFDictionary from the HashMap.
+    // Do this by creating a dictionary with CFDictionaryCreateMutable and then adding the
+    // key-value pairs.
+    // if (sessionStorage)
+    //     sessionStorageDictionary = encodeSessionStorage(*sessionStorage);
+
+    auto stateDictionary = CFDictionaryCreateMutable(
+        kCFAllocatorDefault, 0, &kCFTypeDictionaryKeyCallBacks, &kCFTypeDictionaryValueCallBacks);
```

Comment:
This looks like a memory leak — you'll want to use `adoptCF` here.
```suggestion
    auto stateDictionary = adoptCF(CFDictionaryCreateMutable(
        kCFAllocatorDefault, 0, &kCFTypeDictionaryKeyCallBacks, &kCFTypeDictionaryValueCallBacks));
```

Reply:
Ah, thank you for catching that!